overlord/snapstate/backend/copydata_test.go

```diff
@@ -374,3 +375,43 @@ exit 3
 	err = s.be.CopySnapData(v2, v1, &s.nullProgress)
 	c.Assert(err, ErrorMatches, regexp.QuoteMeta(fmt.Sprintf("cannot copy %s to %s: cp: boom", v1.DataDir(), v2.DataDir())))
 }
+
+func (s *copydataSuite) makeTestDataDir(c *C, snapName string, rev snap.Revision) string {
+	homedir := filepath.Join(s.tempdir, "home", "user1", "snap")
+	homeData := filepath.Join(homedir, snapName, rev.String())
+	err := os.MkdirAll(homeData, 0755)
+	c.Assert(err, IsNil)
+
+	canaryData := []byte("ni ni ni")
+	canaryDataFile := filepath.Join(homeData, fmt.Sprintf("%s-%s.txt", snapName, rev))
+	err = ioutil.WriteFile(canaryDataFile, canaryData, 0644)
+
+	return homeData
+}
+
+func (s *copydataSuite) TestCopyDataOverExistingDir(c *C) {
+	d1 := s.makeTestDataDir(c, "hello", snap.R(1))
+	s.makeTestDataDir(c, "hello", snap.R(2))
```

Comment:
just my opinion, but I feel it would be more natural to make these after the snaps and not before (more like the real world order/timing)